### PR TITLE
`linestring_distance` Header Only API Returns Past-the-End iterator

### LINKED ISSUE #576: [BUG] Pairwise linestring distance should return one past the last written output iterator
According to [development doc](https://github.com/rapidsai/cuspatial/blob/branch-22.08/cpp/doc/libcuspatial_refactoring_guide.md#example-api), to be consistent with `std::transform` API, the method should return the output iterator to one past the last element being written to by the algorithm. Currently, pairwise linestring distance returns `void`:
https://github.com/rapidsai/cuspatial/blob/dbebf67d422104dc47e8ed2ac5a6d2ec75e2e8d0/cpp/include/cuspatial/experimental/linestring_distance.cuh#L61

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR updates `linestring_distance` to return past-the-end iterator to conform to other interfaces. Besides, this PR updates the assumption of `offset` arrays to be arrow compliant format (`N+1` instead of `N`).

closes #576 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cuspatial/experimental/linestring_distance.cuh [hunk 1]
```diff
@@ -59,15 +59,15 @@ namespace cuspatial {
  * "LegacyRandomAccessIterator"
```
Comment (harrism) [inline cpp/include/cuspatial/experimental/linestring_distance.cuh:59]: Add @return doc

**Review by harrism — APPROVED** [review]
One missing @return doc.

**Review by thomcom — APPROVED** [review]

**Comment by isVoid** [comment]
Thanks for the review. I updated the python tests to address the changes to offset format in this PR.

File: cpp/include/cuspatial/experimental/linestring_distance.cuh (hunk 1 above)
Comment (isVoid) [reply cpp/include/cuspatial/experimental/linestring_distance.cuh:60]: don

**Comment by isVoid** [comment]
@gpucibot merge

**Pull request merged**